### ISSUE TITLE
dry-run failed, error: failed to prune fields: failed add back owned items: failed to convert pruned object at version notification.toolkit.fluxcd.io/v1beta2: conversion for notification.toolkit.fluxcd.io/v1beta2, Kind=Alert returned invalid metadata in object at index 1: missing metadata in converted object

### ISSUE THREAD

**Issue by danports** [fluxcd/kustomize-controller#843]
I'm getting this error every reconciliation for a couple of my Kustomization resources and I'm not sure why. The problem started today with no apparent cause; I hadn't pushed any changes to the resource nor upgraded any Flux components recently.

Here's one of the resources reporting this error:
```yaml
apiVersion: notification.toolkit.fluxcd.io/v1beta2
kind: Alert
metadata:
  name: system-deployments
  namespace: flux-system
spec:
  providerRef:
    name: slack
  eventSeverity: info
  eventSources:
    - kind: HelmRelease
      name: '*'
      namespace: kube-system
```

I've tried restarting the Kustomize controller and deleting the Alert resource and neither has made a difference. I don't see anything pertinent in the Kustomize controller logs. I'm using the latest version of all Flux components (v0.35.1 for the Kustomize controller). Anything else I can try?

**Comment by stefanprodan** [comment]
Please see https://github.com/kubernetes/kubernetes/issues/117260#issuecomment-1507226562

**Comment by danports** [comment]
That issue seems to be referencing a regression caused by a refactoring in Kubernetes 1.27, but I'm running 1.26.4. Does this issue also impact 1.26.x, or is what I'm experiencing a different problem?

**Comment by danports** [comment]
My bad - I'm using the same Flux repository for 3 clusters, and 2 of those clusters had been upgraded to 1.27.0, right around the time the errors started. Upgrading those 2 clusters to 1.27.1 seems to have solved the problem. Thanks!

**Comment by runningman84** [comment]
is this problem really fixed? I got the same issue in a 1.29.2 talos cluster:
ustomization/flux-system/traefik-custom-resources dry-run failed: failed to prune fields: failed add back owned items: failed to convert merged object at version kustomize.toolkit.fluxcd.io/v1: .spec.validation: field not declared in sche...

**Comment by danports** [comment]
@runningman84 Based on the `.spec.validation: field not declared in sche...` bit, that sounds like a different problem to me. This specific issue was due to a regression in Kubernetes 1.27.0, corrected in 1.27.1.

**Comment by stefanprodan** [comment]
`.spec.validation` was deprecated years ago, this field must be removed from YAMLs.
